lib/jsonapi/serializer.ex

```diff
@@ -146,29 +149,30 @@ defmodule JSONAPI.Serializer do
     merge_related_links(data, info, remove_links?())
   end
 
-  defp merge_base_links(doc, data, view, conn) do
-    links = Map.merge(%{self: view.url_for(data, conn)}, view.links(data, conn))
+  defp merge_base_links(%{links: links} = doc, data, view, conn) do
+    view_links =
+      data
+      |> view.links(conn)
+      |> Map.merge(links)
+      |> Map.merge(%{self: view.url_for(data, conn)})
 
-    Map.merge(doc, %{links: links})
+    doc
+    |> Map.merge(%{links: view_links})
```

Comment:
We try to avoid pipelines for single function calls. This should be expressed as: `Map.merge(doc, %{links: view_links})`

Reply:
Noted, also fixed.